### PR TITLE
Relax (0.5x) reverse-penalty for HH with inter-points

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd/issues/18788

File: OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java [hunk 1]
```diff
@@ -830,6 +830,9 @@ public RouteCalcResult searchRoute(final RoutingContext ctx, LatLon start, LatLo
 			HHRoutePlanner<NetworkDBPoint> routePlanner = HHRoutePlanner.create(ctx, null);
 			HHNetworkRouteRes r = null;
 			Double dir = ctx.config.initialDirection ;
+			if (targets.size() > 1) {
+				ctx.config.penaltyForReverseDirection /= 2; // relax reverse-penalty in case of inter-points
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java:834]: 1) not correct for 1st point 2) not applicable for other type of routings ?

Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java:834]: 1. It was correct and now is clarified additionally.

3. Yes, this penalty is basically applied for inter-points routes for HH-engine only. If you decide to have the same for BRP-engine, I'll implement it.

Comment (vshcherb) [reply OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java:834]: It's not correct, for the first point. It should be applied only to intermediates and revert later

Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java]: Fixed: 1) inter-points only, from second target; 2) reverted once after calculateHHRoute()

**Pull request merged**